server/plugin.go

```diff
@@ -412,37 +412,26 @@ func (p *Plugin) syncUsers() {
 				Username:  username,
 			}
 
-			newUser, appErr := p.API.CreateUser(newMMUser)
-			if appErr != nil {
-				if strings.Contains(appErr.Error(), "account with that username already exists") {
-					value, sErr := p.API.KVGet(msDisplayName)
-					if sErr != nil {
-						p.API.LogError("Unable to get value from KV store", "error", sErr.Error())
+			var newUser *model.User
+			for {
+				newUser, appErr = p.API.CreateUser(newMMUser)
+				if appErr != nil {
+					newUsername := p.CheckAndGetUsername(appErr, newMMUser.Username, userSuffixID)
```

Comment:
This approach of passing an error to handle it, I don't think is the right approach, I think here I prefer a bit of repetition than this kind of encapsulation.

I would copy the content of the function directly here in the places where we are calling this function. If you want something to encapsulate, I would suggest to encapsulate something like `ensureCreateUserWithSuffix` or something like that. That returns the new user and an error in case the reason to fail is not the duplicated user.

You are swallowing here any error reason without failing. For example, my expected behavior is, if I can't create a user because, for example, a temporary problem accessing the database, I don't want you to create a different user with the next suffix available, I want you to fail and give that error.